frontend/tests/controllers/BadgesTest.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/**
+ * VirtualContestTest
+ *
+ * @author carlosabcs
+ */
+
+class BadgesTest extends OmegaupTestCase {
+    const OMEGAUP_BADGES_ROOT = OMEGAUP_ROOT . '/badges';
+    const MAX_BADGE_SIZE = 20 * 1024;
+    const ICON_FILE = 'icon.svg';
+    const LOCALIZATIONS_FILE = 'localizations.json';
+    const QUERY_FILE = 'query.sql';
+    const TEST_FILE = 'test.json';
+
+    public function testAllBadges() {
+        $aliases = glob(static::OMEGAUP_BADGES_ROOT . '/*', GLOB_ONLYDIR);
+        foreach ($aliases as $alias) {
+            // Gets the string after last '/'
+            $badge = substr($alias, strrpos($alias, '/') + 1);
+
+            $icon_path = $alias . '/' . static::ICON_FILE;
```

Comment:
nit: las variables deben ir en camelCase:

```suggestion
            $iconPath = "${badgePath}/" . static::ICON_FILE;
```